ambry-router/src/test/java/com.github.ambry.router/GetOperationTest.java

```diff
@@ -0,0 +1,474 @@
+        correlationIdToGetOperation.size());
+
+    List<ResponseInfo> responses = sendAndWaitForResponses(requestListToFill);
+    for (ResponseInfo response : responses) {
+      op.handleResponse(response);
+      if (op.isOperationComplete()) {
+        break;
+      }
+    }
+    Assert.assertTrue("Operation should be complete at this time", op.isOperationComplete());
+    assertSuccess();
+  }
+
+  /**
+   * Test the case where all requests time out within the GetOperation.
+   * @throws Exception
+   */
+  @Test
+  public void testRouterRequestTimeoutAllFailure()
+      throws Exception {
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture,
+            operationCallback, time);
+    requestRegistrationCallback.requestListToFill = new ArrayList<>();
+    op.poll(requestRegistrationCallback);
+    while (!op.isOperationComplete()) {
+      time.sleep(routerConfig.routerRequestTimeoutMs + 1);
+      op.poll(requestRegistrationCallback);
+    }
+    // At this time requests would have been created for all replicas, as none of them were delivered,
+    // and cross-colo proxying is enabled by default.
+    Assert.assertEquals("Must have attempted sending requests to all replicas", replicasCount,
+        correlationIdToGetOperation.size());
+    Assert.assertTrue("Operation should be complete at this time", op.isOperationComplete());
+    Assert.assertTrue(operationFuture.isDone());
+    Assert.assertEquals(operationException.get(), op.getOperationException());
+    RouterException routerException = (RouterException) operationException.get();
+    Assert.assertEquals(RouterErrorCode.OperationTimedOut, routerException.getErrorCode());
+  }
+
+  /**
+   * Test the case where all requests time out within the NetworkClient.
+   * @throws Exception
+   */
+  @Test
+  public void testNetworkClientTimeoutAllFailure()
+      throws Exception {
+    GetBlobInfoOperation op =
+        new GetBlobInfoOperation(routerConfig, mockClusterMap, responseHandler, blobIdStr, operationFuture,
+            operationCallback, time);
+    ArrayList<RequestInfo> requestListToFill = new ArrayList<>();
+    requestRegistrationCallback.requestListToFill = requestListToFill;
+
+    while (!op.isOperationComplete()) {
+      op.poll(requestRegistrationCallback);
+      List<ResponseInfo> responses = sendAndWaitForResponses(requestListToFill);
+      requestListToFill.clear();
+      for (ResponseInfo response : responses) {
+        ResponseInfo fakeResponse =
+            new ResponseInfo(response.getRequest(), NetworkClientErrorCode.NetworkError, response.getResponse());
+        op.handleResponse(fakeResponse);
+        if (op.isOperationComplete()) {
+          break;
+        }
+      }
+    }
+
+    Assert.assertEquals(replicasCount, correlationIdToGetOperation.size());
```

Comment:
Having a message here would certainly help understand why the test is trying to do what it is doing.